### ISSUE TITLE
ValidSetSwing - OPTION check is wrong

### ISSUE THREAD

**Issue by pappde** [pappde/bmai#18]
OBSERVED
The compiler shows this warning
````
Warning	C4804	'>': unsafe use of type 'bool' in operation	
````

For this code
````
		if (d->HasProperty(BME_PROPERTY_OPTION))
		{
			if (_move.m_option_die[i]>1)
				return false;
		}
````

m_option is a BitArray and it returns true or false, so the "return false" will never run.

EXPECTED:
No compiler warning, and this code returns false in the given case appropriately.

**Comment by hamstercrack** [comment]
There's a comment on the function that suggests that it might not effectively matter:

````
// drp071505 - OPTION check was wrong (not checking any dice, but probably didn't matter)
````

**Comment by pappde** [comment]
fixed in commit 3726079
